src/analyses/interval_domain.h

```diff
@@ -56,26 +56,40 @@ class interval_domaint:public ai_domain_baset
   }
 
   // no states
-  void make_bottom() final
+  void make_bottom() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=true;
   }
 
   // all states
-  void make_top() final
+  void make_top() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=false;
   }
 
-  void make_entry() final
+  void make_entry() final override
   {
     make_top();
   }
 
+  bool is_bottom() const override final
+  {
+    // This invariant should hold but is not correctly enforced at the moment.
+    // DATA_INVARIANT(!bottom || (int_map.empty() && float_map.empty()),
+    //                "If the domain is bottom the value maps must be empty");
+
+    return bottom;
+  }
+
+  bool is_top() const override final
+  {
+    return !bottom && int_map.empty() && float_map.empty();
```

Comment:
with the above, is this then correct?

Reply:
That is an interesting question.  It seems to work for the use cases it has currently been deployed for but ... TBH ... I am kinda suspicious about this domain and would not recommend it's use.